packages/date-picker/test/value-commit.common.js

```diff
@@ -0,0 +1,397 @@
+      await waitForOverlayRender();
+      valueChangedSpy.resetHistory();
+      validateSpy.resetHistory();
+      changeSpy.resetHistory();
+    });
+
+    describe('input cleared with Backspace', () => {
+      beforeEach(async () => {
+        datePicker.inputElement.select();
+        await sendKeys({ press: 'Backspace' });
+      });
+
+      it('should commit on blur', () => {
+        datePicker.blur();
+        expectValueCommit('');
+      });
+
+      it('should commit on Enter', async () => {
+        await sendKeys({ press: 'Enter' });
+        expectValueCommit('');
+      });
+
+      it('should commit on Escape', async () => {
+        await sendKeys({ press: 'Escape' });
+        expectValueCommit('');
+      });
+
+      it('should commit on close with Escape', async () => {
+        await sendKeys({ press: 'ArrowDown' });
+        await waitForOverlayRender();
+        await sendKeys({ press: 'Escape' });
+        expectValueCommit('');
+      });
+    });
+  });
+
+  describe('unparsable input entered', () => {
+    beforeEach(async () => {
+      await sendKeys({ type: 'foo' });
+      await waitForOverlayRender();
+    });
+
+    it('should not commit but validate on Enter', async () => {
+      await sendKeys({ press: 'Enter' });
+      expectValidationOnly();
+      expect(datePicker.inputElement.value).to.equal('foo');
+    });
+
+    it('should not commit but validate on close with outside click', () => {
+      outsideClick();
+      expectValidationOnly();
+      expect(datePicker.inputElement.value).to.equal('foo');
+    });
+
+    it('should revert on close with Escape', async () => {
+      await sendKeys({ press: 'Escape' });
+      expectNoValueCommit();
+      expect(datePicker.inputElement.value).to.equal('');
+    });
+  });
+
+  describe('unparsable input committed', () => {
+    beforeEach(async () => {
+      await sendKeys({ type: 'foo' });
+      await sendKeys({ press: 'Enter' });
+      await waitForOverlayRender();
+      validateSpy.resetHistory();
+    });
+
+    describe('input cleared with Backspace', () => {
+      beforeEach(async () => {
+        datePicker.inputElement.select();
+        await sendKeys({ press: 'Backspace' });
+      });
+
+      it('should not commit but validate on Enter', async () => {
+        await sendKeys({ press: 'Enter' });
+        expectValidationOnly();
+      });
+
+      it('should not commit but validate on outside click', () => {
+        outsideClick();
+        expectValidationOnly();
+      });
+    });
+  });
+
+  describe('overlay date focused', () => {
+    beforeEach(async () => {
+      // Open the dropdown.
+      await sendKeys({ press: 'ArrowDown' });
+      await waitForOverlayRender();
+      // Focus yesterday's date.
+      await sendKeys({ press: 'ArrowLeft' });
+      await waitForScrollToFinish(datePicker._overlayContent);
+    });
+
+    it('should commit on focused date selection with click', async () => {
```

Comment:
nit: not needed `async`

```suggestion
    it('should commit on focused date selection with click', () => {
```